qiskit_nature/results/utils/protein_decoder.py

```diff
@@ -0,0 +1,109 @@
+    ) -> None:
+        """
+        Args:
+            best_sequence: Will be the sequence decoded.
+            side_chain_hot_vector: boolean list with the position of the side chains
+            fifth_bit: True if the fifth bit has defaulted to 1.
+        """
+        self._best_sequence = best_sequence
+        self._side_chain_hot_vector = side_chain_hot_vector
+        self._fifth_bit = fifth_bit
+        self._main_chain_length = len(side_chain_hot_vector)
+        self.main_turns = self._get_main_turns()
+        self.side_turns = self._get_side_turns()
+
+    def _bitstring_to_turns(self, bitstring: str) -> List[int]:
+        """
+        Takes a bitstring encoding the turns of a chain and returns the turns as a list of int.
+
+        Args:
+            bitstring: string containing the encoded information.
+        Returns:
+            A list of integers decoding the bitstring.
+        """
+        bitstring = bitstring[::-1]
+        encoding = {"00": 0, "01": 1, "10": 2, "11": 3}
+        length_turns = len(bitstring) // 2
+        return [encoding[bitstring[2 * i : 2 * (i + 1)]] for i in range(length_turns)]
+
+    def _split_bitstring(self) -> Tuple[int, int]:
+        """Returns the amount of bits in the compact solution corresponding
+        to each property they encode."""
+        n_qbits_encoding_main_turns = 2 * (self._main_chain_length - 3) - (self._fifth_bit)
+        n_qbits_encoding_side_turns = 2 * sum(self._side_chain_hot_vector)
+        return n_qbits_encoding_main_turns, n_qbits_encoding_side_turns
+
+    def _get_main_turns(self) -> List[int]:
+        """
+        Returns the list of turns for the molecule corresponding to best_sequence.
+        The first element of the list corresponds to the turn of the second aminoacid in the peptide.
+        Returns:
+                A list of integers representing the sequence of turns on the molecule
+
+        Notes:
+                The bitstring will end in 0010 corresponding to turn1=(01) and turn2=(00)
+                If the second bead doesn't have a side chain the 6th bit
+                can be set to 1 without loss of generality.
+                In that case index (5) will belong to the list of unused qubits.
+                The amount of qubits needed to encode the turns will be 2(N-3) - 1
+                if no side chain on second main bead or 2(N-3) otherwise.
+        """
+
+        main_turns_bitstring = self._best_sequence[-self._split_bitstring()[0] :] + "0010"
+
+        if self._fifth_bit:
+            main_turns_bitstring = main_turns_bitstring[:-5] + "1" + main_turns_bitstring[-5:]
+
+        return self._bitstring_to_turns(main_turns_bitstring)
+
+    def _get_side_turns(self) -> List[Optional[int]]:
+        """
+        Returns the list of turns from the main bead corresponding to the side chains.
+        None corresponds to no side chain from that main bead.
```

Comment:
I think we usually try to format code in docs, by surrounding by double ` symbols., I believe. Please check other places for variable names and similar.